template-manager-cmtemplate/src/test/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/CmHostGroupRoleConfigProvidersTest.java

```diff
@@ -0,0 +1,238 @@
+package com.sequenceiq.cloudbreak.cmtemplate.configproviders;
+
+import static com.sequenceiq.cloudbreak.cmtemplate.configproviders.ConfigUtils.config;
+import static java.util.Optional.ofNullable;
+import static org.junit.Assert.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Spy;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import com.cloudera.api.swagger.model.ApiClusterTemplateConfig;
+import com.cloudera.api.swagger.model.ApiClusterTemplateRoleConfigGroup;
+import com.sequenceiq.cloudbreak.cmtemplate.CmHostGroupRoleConfigProvider;
+import com.sequenceiq.cloudbreak.cmtemplate.CmHostGroupRoleConfigProviders;
+import com.sequenceiq.cloudbreak.cmtemplate.CmTemplateProcessor;
+import com.sequenceiq.cloudbreak.cmtemplate.configproviders.hdfs.HdfsVolumeConfigProvider;
+import com.sequenceiq.cloudbreak.cmtemplate.configproviders.yarn.YarnVolumeConfigProvider;
+import com.sequenceiq.cloudbreak.common.type.InstanceGroupType;
+import com.sequenceiq.cloudbreak.template.TemplatePreparationObject;
+import com.sequenceiq.cloudbreak.template.TemplatePreparationObject.Builder;
+import com.sequenceiq.cloudbreak.template.model.GeneralClusterConfigs;
+import com.sequenceiq.cloudbreak.template.views.BlueprintView;
+import com.sequenceiq.cloudbreak.template.views.HostgroupView;
+import com.sequenceiq.cloudbreak.util.FileReaderUtils;
+
+@RunWith(MockitoJUnitRunner.class)
+public class CmHostGroupRoleConfigProvidersTest {
+
+    @InjectMocks
+    private final CmHostGroupRoleConfigProviders underTest = new CmHostGroupRoleConfigProviders();
+
+    @Spy
```

Comment:
What is the reason for using a spy? What functionality do we want to stub on it?

Reply:
I need it to be injected into the test subject (see `@InjectMocks`), but didn't want to `@Mock` it.  Using `@Spy` is one way to achieve this.  If you know a better way, please share. :)

Reply:
Ah. For injection purposes that indeed makes sense. Otherwise it could be also set explicitly into the respective field using SpringTest's `ReflectionTestUtils.setField()`.